// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.password_manager.core.browser.proto";

option optimize_for = LITE_RUNTIME;

package password_manager;

import "chrome/browser/password_manager/android/protos/password_with_local_data.proto";
import "chrome/browser/password_manager/android/protos/password_info.proto";

// Response to a request sent to Google Mobile Services to request a list of
// passwords.
message ListPasswordsResult {
  // The list of password entries and corresponding additional info.
  repeated PasswordWithLocalData password_data = 1;
}

// Response to a request sent to Google Mobile Services to request a list of
// all passwords with passwords UI info.
message ListPasswordsWithUiInfoResult {
  // Password with additional UI info.
  message PasswordWithUiInfo {
    // Password with local data itself.
    optional PasswordWithLocalData password_data = 1;

    // PasswordInfo with additional UI information. Absent if there is no
    // affiliation data for the sign on realm of the password in the ChromeSync
    // database. This should happen rarely for http://, https://, android:// and
    // federation:// credentials. For other types of PasswordSpecificsData like
    // proxy auth this info will be absent.
    optional PasswordInfo ui_info = 2;
  }

  // List of passwords with UI info.
  repeated PasswordWithUiInfo passwords_with_ui_info = 1;
}
